charts/thanos/templates/store-gateway/pdb.yaml

```diff
@@ -1,4 +1,4 @@
-{{- if .Values.storeGateway.podDisruptionBudget.enabled -}}
+{{- if and .Values.storeGateway.podDisruptionBudget.enabled .Values.storeGateway.enabled -}}
```

Comment:
I'd prefer the `.Values.storeGateway.enabled` be the first clause for readability.